## Cell 1 — Import libraries

In [1]:
import random
import numpy as np
import pandas as pd


## Cell 2 — Enter your topics

In [2]:
topics = [
    "Compare the importance of self-reliance and adaptability in healthcare.",
    "Evaluate management consulting in addressing conflicts within marketing.",
    "Discuss the role of self-reliance in achieving success in software engineering.",
]

print(f'✅ You have {len(topics)} topic(s):')
for i, t in enumerate(topics):
    print(f'  {i+1}. {t}')

✅ You have 3 topic(s):
  1. Compare the importance of self-reliance and adaptability in healthcare.
  2. Evaluate management consulting in addressing conflicts within marketing.
  3. Discuss the role of self-reliance in achieving success in software engineering.


## Cell 3 — Essay generator

In [6]:
# Essay templates — designed to be ambiguous and cause disagreement
def generate_essay(topic):
    templates = [
        # Template 1 — philosophical / contradictory
        (
            "The question of {t} is, at its core, a paradox. "
            "Those who champion rigid principles claim they form the bedrock of progress — "
            "without them, systems collapse into chaos. "

        ),
        # Template 2 — metaphorical
        (
            "Think of {t} as a river and a dam. "
            "The river — wild and indifferent — represents raw potential. "
            "The dam channels it, makes it useful, but at a cost: the valley upstream drowns. "

        ),
        # Template 3 — dialectical / thesis-antithesis
        (
            "Thesis: {t} demands clarity, structure, and measurable outcomes. "
            "Antithesis: it demands empathy, ambiguity, and irreducible human judgment. "
            "Synthesis: neither alone is enough — yet combining them breeds its own contradictions. "
            "Practitioners who rely entirely on data risk missing what data was never designed to capture. "

        ),
    ]
    chosen = random.choice(templates)
    return chosen.format(t=topic.lower().rstrip('.'))

print('✅ Essay generator ready!')

✅ Essay generator ready!


## Cell 4 — Scoring system

In [7]:
def score_essay(essay):
    criteria = ['Introduction', 'Logic', 'Emotion', 'Philosophy', 'Conclusion']
    judge_scores = {}

    for c in criteria:
        # 5 judges each give a score from 0 to 10
        scores = [round(random.uniform(2, 10), 1) for _ in range(5)]
        judge_scores[c] = {
            'avg'  : round(np.mean(scores), 2),
            'stdev': round(np.std(scores), 2),
        }

    avgs   = [v['avg']   for v in judge_scores.values()]
    stdevs = [v['stdev'] for v in judge_scores.values()]

    avg_quality     = round(np.mean(avgs), 2)
    avg_disagreement= round(np.mean(stdevs), 2)
    min_disagreement= round(np.min(stdevs), 2)

    # Final score — higher means more disagreement between judges
    final = round(avg_disagreement * min_disagreement * (9 - avg_quality), 4)

    return judge_scores, avg_quality, avg_disagreement, final

print('✅ Scoring system ready!')

✅ Scoring system ready!


## Cell 5 — Run the test

In [8]:
results = []

for topic in topics:
    essay = generate_essay(topic)
    judge_data, avg_q, avg_d, final = score_essay(essay)
    results.append({
        'topic'      : topic,
        'essay'      : essay,
        'judge_data' : judge_data,
        'avg_quality': avg_q,
        'avg_disagree': avg_d,
        'final_score': final,
    })

print('✅ Done! Scroll down to see results.')

✅ Done! Scroll down to see results.


## Cell 6 — results

In [9]:
for i, r in enumerate(results):
    print('=' * 60)
    print(f'Essay {i+1}: {r["topic"]}')
    print('=' * 60)

    print('\n📄 ESSAY:')
    print(r['essay'])

    print('\n📊 JUDGE SCORES (avg ± disagreement):')
    for criterion, vals in r['judge_data'].items():
        bar = '█' * int(vals['avg'])
        print(f"  {criterion:<15} {bar:<10} avg={vals['avg']}  disagreement={vals['stdev']}")

    print(f"\n🎯 FINAL SCORE : {r['final_score']}")
    print(f"   Avg quality  : {r['avg_quality']} / 10")
    print(f"   Avg disagree : {r['avg_disagree']} (higher = judges disagree more)")
    print()

Essay 1: Compare the importance of self-reliance and adaptability in healthcare.

📄 ESSAY:
Think of compare the importance of self-reliance and adaptability in healthcare as a river and a dam. The river — wild and indifferent — represents raw potential. The dam channels it, makes it useful, but at a cost: the valley upstream drowns. 

📊 JUDGE SCORES (avg ± disagreement):
  Introduction    █████      avg=5.94  disagreement=2.95
  Logic           █████      avg=5.5  disagreement=2.17
  Emotion         ███████    avg=7.38  disagreement=2.02
  Philosophy      ██████     avg=6.42  disagreement=2.93
  Conclusion      █████      avg=5.9  disagreement=2.05

🎯 FINAL SCORE : 13.5409
   Avg quality  : 6.23 / 10
   Avg disagree : 2.42 (higher = judges disagree more)

Essay 2: Evaluate management consulting in addressing conflicts within marketing.

📄 ESSAY:
Thesis: evaluate management consulting in addressing conflicts within marketing demands clarity, structure, and measurable outcomes. Antithesi

## Cell 7 — Summary table

In [10]:
df = pd.DataFrame([{
    'Topic'        : r['topic'][:50] + '...',
    'Avg Quality'  : r['avg_quality'],
    'Avg Disagree' : r['avg_disagree'],
    'Final Score'  : r['final_score'],
} for r in results])

# Highlight best and worst
display(df.style
    .highlight_max(subset=['Final Score', 'Avg Disagree'], color='lightgreen')
    .highlight_min(subset=['Final Score', 'Avg Disagree'], color='#ffcccc')
)

best = df['Final Score'].idxmax()
print(f'\n🏆 Best essay: Essay {best+1}')
print(f'   Topic: {results[best]["topic"]}')
print(f'   Score: {results[best]["final_score"]}')

,Topic,Avg Quality,Avg Disagree,Final Score
0,Compare the importance of self-reliance and adapta...,6.230000,2.420000,13.540900
1,Evaluate management consulting in addressing confl...,5.840000,1.960000,6.069700
2,Discuss the role of self-reliance in achieving suc...,5.920000,2.130000,9.119000



🏆 Best essay: Essay 1
   Topic: Compare the importance of self-reliance and adaptability in healthcare.
   Score: 13.5409


## Cell 8 — Save to CSV

In [11]:
submission = pd.DataFrame({
    'id'   : range(1, len(results) + 1),
    'essay': [r['essay'] for r in results]
})

submission.to_csv('submission.csv', index=False)
print('✅ Saved as submission.csv')
print('   (Find it in the Files tab on the left sidebar)')
display(submission)

✅ Saved as submission.csv
   (Find it in the Files tab on the left sidebar)


,id,essay
0,1,Think of compare the importance of self-relian...
1,2,Thesis: evaluate management consulting in addr...
2,3,Think of discuss the role of self-reliance in ...


Based on the disagreement results (Avg disagree : 2.42, 1.96, 2.13) the model still produces only moderate disagreement among LLM judges. Although some essays (e.g., Essay 1) achieved higher disagreement scores, others remain relatively low. This indicates that the current model is not consistently effective in exploiting differences between LLM judges. Therefore, further training and improvement are required to generate more conflicting, ambiguous, and stylistically diverse essays to maximize disagreement.